src/main/java/com/aiykr/iquais/service/impl/UserServiceImpl.java

```diff
@@ -23,40 +25,67 @@
-
-            //Create Guardian
-            UserDAO guardianDAO = new UserDAO();
-            guardianDAO.setFirstName(postUserDTO.getGuardianFirstName());
-            guardianDAO.setLastName(postUserDTO.getGuardianLastName());
-            guardianDAO.setEmail(postUserDTO.getGuardianEmail());
-            guardianDAO.setType(UserType.GUARDIAN.name());
-
-            userRepository.save(guardianDAO);
-        } catch (Exception ex) {
-            throw new ResponseExceptionHandler("Exception occured while generating and storing in DB");
+            String studentEmail = postUserDTO.getEmail();
+            log.info("Student Email: {}", studentEmail);
+            List<UserDAO> studentUsers = userRepository.findAllByEmail(studentEmail);
+            log.info("Student Email from DB: {}", studentUsers);
+            boolean studentExists = studentUsers.stream().anyMatch(user -> user.getEmail() != null && user.getEmail().equals(studentEmail));
+            if (studentExists) {
+                log.info("Student Email {} already present in our DB", studentUsers.get(0).getEmail());
+                throw new IquaisException(HttpStatus.CONFLICT,"IQ002","Student Email already present in our DB");
+            } else {
+                // Create Student
+                UserDAO studentDAO = new ModelMapper().map(postUserDTO, UserDAO.class);
+                studentDAO.setType(UserType.STUDENT.name());
+                userRepository.save(studentDAO);
+                log.info("Save New Student: {}", studentDAO);
+            }
+
+            // Check if the guardian is already present by email address
+            String guardianEmail = postUserDTO.getGuardianEmail();
+            log.info("Guardian Email: {}", guardianEmail);
+            List<UserDAO> guardianUsers = userRepository.findAllByEmail(guardianEmail);
+            log.info("Guardian Email from DB: {}", guardianUsers);
+            boolean guardianExists = guardianUsers.stream().anyMatch(user ->
+                    user.getEmail() != null && user.getEmail().equals(guardianEmail)
+            );
+            if (guardianExists) {
+                log.info("Guardian Email {} already present in our DB", guardianUsers.get(0).getEmail());
+            } else {
+                // Create Guardian
+                UserDAO guardianDAO = new UserDAO();
+                guardianDAO.setFirstName(postUserDTO.getGuardianFirstName());
+                guardianDAO.setLastName(postUserDTO.getGuardianLastName());
+                guardianDAO.setEmail(postUserDTO.getGuardianEmail());
+                guardianDAO.setType(UserType.GUARDIAN.name());
+                userRepository.save(guardianDAO);
+                log.info("Save New Guardian: {}", guardianDAO);
+            }
+        } catch (IquaisException iqEx) {
+            throw iqEx;
+        }
+        catch (Exception ex) {
+            log.error("An exception occurred: {}", ex.getMessage(), ex);
+            ex.getStackTrace();
```

Comment:
remove this

Reply:
did understand as the line is not mentioned clearly